File: Sources/Contracts/AICommitServicing.swift
// MARK: - AICommitServicing Protocol
// Контракт для AI-генерации сообщений коммитов.
// macOS 14+, Swift 5.10

import Foundation

// MARK: - Protocol

/// Contract for AI-powered commit message generation.
///
/// Implementations call a remote LLM to produce a concise conventional
/// commit message from a staged git diff.
protocol AICommitServicing: Sendable {
    /// Generate a conventional commit message for the given diff text.
    ///
    /// - Parameter diff: The full staged diff (may be truncated internally).
    /// - Returns: A concise commit message string.
    /// - Throws: ``AICommitServiceError`` when the API key is missing,
    ///   the request fails, or the response cannot be parsed.
    func generateCommitMessage(for diff: String) async throws -> String
}

File: shared/src/commonMain/kotlin/studio/vibe/shared/core/common/AICommitServicing.kt
package studio.vibe.shared.core.common

import kotlin.coroutines.cancellation.CancellationException
import studio.vibe.shared.core.common.AICommitServiceError

interface AICommitServicing {
    /**
     * Generates a commit message from [diff] using an LLM.
     *
     * @throws AICommitServiceError if the API key is missing, the request fails,
     *   or the response cannot be parsed.
     * @throws CancellationException if the coroutine is cancelled while awaiting
     *   the network response.
     */
    @Throws(AICommitServiceError::class, CancellationException::class)
    suspend fun generateCommitMessage(diff: String): String
}
